app/src/main/java/com/example/nigeriatelemedicineapp/registerpatient/RegisterPatientActivity.kt

```diff
@@ -73,4 +76,75 @@ class RegisterPatientActivity : AppCompatActivity() {
         )
     }
 
+    private fun handleStatus(status: Status?) {
+        when (status) {
+
+            Status.EMPTY_FIRST_NAME -> {
+                binding.firstName.requestFocus()
+                binding.firstName.setError("Please enter your first name")
+            }
+
+            Status.EMPTY_LAST_NAME -> {
+                binding.lastName.requestFocus()
+                binding.lastName.setError("Please enter your last name")
+            }
+
+            Status.EMPTY_GENDER -> Toast.makeText(this, "Please select your gender", Toast.LENGTH_SHORT).show()
+
+            Status.EMPTY_PHONE -> {
+                binding.phoneNumber.requestFocus()
+                binding.phoneNumber.setError("Please Enter your phone number")
+            }
+
+            Status.EMPTY_DOB -> {
+                binding.DOB.requestFocus()
+                binding.DOB.setError("Please enter your DOB")
+            }
+
+            Status.SUCCESS -> startAnimations()
+        }
+    }
+
+    fun hideKeyboard(activity: Activity = this) {
+        val imm = activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
+        //Find the currently focused view, so we can grab the correct window token from it.
+        var view = activity.currentFocus
+        //If no view currently has focus, create a new one, just so we can grab a window token from it
+        if (view == null) {
+            view = View(activity)
+        }
+        imm.hideSoftInputFromWindow(view.windowToken, 0)
+    }
+
+    private fun startAnimations() {
+        hideKeyboard()
+        binding.CardView.visibility = View.INVISIBLE
+        binding.text.visibility = View.INVISIBLE
+        binding.progressbar.visibility = View.VISIBLE
+        binding.loadingtext.visibility = View.VISIBLE
+    }
+
+    fun getSuccessDialog(): AlertDialog {
+        val builder = AlertDialog.Builder(this)
+        builder.setCancelable(false)
+        builder.setTitle("Success")
+        builder.setMessage(R.string.SuccessDialogText)
+        builder.setPositiveButton("back to dashboard") { dialog, which ->
+            finish()
+            startActivity(Intent(this, DashBoardActivity::class.java))
+        }
+        return builder.create()
+    }
+
+    fun getFailureDialog(): AlertDialog {
+        val builder = AlertDialog.Builder(this)
+        builder.setCancelable(false)
+        builder.setTitle("Error")
+        builder.setMessage(R.string.FailureDialogText)
```

Comment:
Can we get some more specific dialog messages here? We should at least be able to differentiate between network connectivity errors (could not connect), server-side errors (500 error), and client-side errors.